mmv1/templates/terraform/custom_flatten/network_services_timeout_mirror.go.erb

```diff
@@ -13,9 +13,22 @@
 	# limitations under the License.
 -%>
 func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *Config) interface{} {
-	out := make(map[string]interface{})
-	out["connect_timeout"] = d.Get("timeout.0.connect_timeout")
-	out["max_attempts_timeout"] = d.Get("timeout.0.max_attempts_timeout")
-	out["responseTimeout"] = d.Get("timeout.0.responseTimeout")
+	out := make(map[string]string)
+
+	if v == nil {
+		return nil
+	}
+
+	in := v.(map[string]interface{})
+	for _, names := range []struct{ in, out string }{
+		{"connectTimeout", "connect_timeout"},
+		{"maxAttemptsTimeout", "max_attempts_timeout"},
+		{"responseTimeout", "response_timeout"},
+		{"readTimeout", "read_timeout"},
+	} {
+		if e, ok := in[names.in]; ok {
+			out[names.out] = e.(string)
+		}
+	}
```

Comment:
The readability isn't ideal for this. Preference for doing this iterative instead of in a loop. It's not immediately clear whats going on.

Reply:
What do you mean by "doing this iterative"?

I could flatten everything out with a bunch of code like:
```
if e, ok := in["connectTimeout"]; ok {
   out["connect_timeout" = e.(string)
}
if e, ok := in["maxAttemptsTimeout"]; ok {
   out["max_attempts_timeout" = e.(string)
}
...
```

Reply:
I unrolled the loop.